tests/test_pages.py

```diff
@@ -115,7 +117,7 @@ def test_menu_item(selenium, user, url):
         EC.presence_of_element_located((By.ID, 'account_menu'))
     )
     assert account_menu.text == user.name, 'Logged in user shows the correct name'
-    assert selenium.current_url == url, 'Correct page is loaded'
+    assert url in selenium.current_url, 'Correct page is loaded'
```

Comment:
I think the order of get arguments may change so I was thinking about this:
```python
from typing import Union, KeysView, List
from urllib.parse import ParseResult, urlparse, parse_qs, urlencode

import pytest


def filtered_url_query(url: str, allowed_query_params: Union[List, KeysView]) -> ParseResult:
    parsed_url = urlparse(url)
    query = parse_qs(parsed_url.query)
    reduced_query = {key: value for key, value in query.items() if key in allowed_query_params}
    new_query = urlencode(reduced_query, doseq=True)
    return parsed_url._replace(query=new_query)


@pytest.mark.parametrize('actual_url,expected_url', [
    ['http://example.com', 'http://example.com'],                                            
    ['http://example.com/path', 'http://example.com/path'],                                  
    ['http://example.com/path?param=true', 'http://example.com/path'],                       
    ['http://example.com/path?param=true', 'http://example.com/path?param=true'],            
    ['http://example.com?param=true', 'http://example.com'],                                 
    ['http://example.com?param=true', 'http://example.com?param=true'],                      
    ['http://example.com?param=true&other=false', 'http://example.com?param=true'],          
    ['http://example.com?param=true&other=false#hash', 'http://example.com?param=true#hash'],
])
def test_url_matches(actual_url: str, expected_url: str) -> None:
    expected = urlparse(expected_url)
    allowed_query_params = parse_qs(expected.query).keys()
    actual = filtered_url_query(actual_url, allowed_query_params)
    assert actual == expected


@pytest.mark.parametrize('actual_url,expected_url', [
    ['https://example.com', 'http://example.com'],
    ['http://example.com/path', 'http://example.com'],
    ['http://example.com', 'http://example.com/path'],
    ['http://example.com#hash', 'http://example.com'],
    ['http://example.com', 'http://example.com#hash'],
    ['http://example.com', 'http://example.com?param=true'],
    ['http://example.com?param=first&param=second', 'http://example.com?param=true'],
    ['http://example.com?param=true', 'http://example.com?param=true&other=false'],
])
def test_url_does_not_matches(actual_url: str, expected_url: str) -> None:
    expected = urlparse(expected_url)
    allowed_query_params = parse_qs(expected.query).keys()
    actual = filtered_url_query(actual_url, allowed_query_params)
    assert actual != expected
```

I added tests and data types to ensure it's correct. Since Smoker is Python 3+ I think we can rely on them.

Reply:
Given the size of that chunk, I think it'd be better for you to open a PR with the changes and I can close this in favor of that.

Reply:
I was thinking exactly the same.